bqskit/ir/gates/parameterized/rxx.py

```diff
@@ -4,54 +4,94 @@
 import numpy as np
 import numpy.typing as npt
 
-from bqskit.ir.gates.qubitgate import QubitGate
+from bqskit.ir.gates.quditgate import QuditGate
 from bqskit.qis.unitary.differentiable import DifferentiableUnitary
-from bqskit.qis.unitary.optimizable import LocallyOptimizableUnitary
 from bqskit.qis.unitary.unitary import RealVector
 from bqskit.qis.unitary.unitarymatrix import UnitaryMatrix
 from bqskit.utils.cachedclass import CachedClass
+from bqskit.utils.typing import is_integer
 
 
 class RXXGate(
-    QubitGate,
+    QuditGate,
     DifferentiableUnitary,
-    LocallyOptimizableUnitary,
     CachedClass,
 ):
     """
     A gate representing an arbitrary rotation around the XX axis.
 
-    It is given by the following parameterized unitary:
-
-    .. math::
-
-        \\begin{pmatrix}
-        \\cos{\\frac{\\theta}{2}} & 0 & 0 & -\\sin{\\frac{\\theta}{2}}i \\\\
-        0 & \\cos{\\frac{\\theta}{2}} & -\\sin{\\frac{\\theta}{2}}i & 0 \\\\
-        0 & -\\sin{\\frac{\\theta}{2}}i & \\cos{\\frac{\\theta}{2}} & 0 \\\\
-        -\\sin{\\frac{\\theta}{2}}i & 0 & 0 & \\cos{\\frac{\\theta}{2}} \\\\
-        \\end{pmatrix}
+    __init__() arguments:
+            num_levels : int
+                Number of levels in each qudit (d).
+            level_1,level_2, level_3, level_4: int
+                The levels on which to apply the X gates (0...d-1).
+    get_unitary arguments:
+            param: float
+            The angle by which to rotate
     """
 
     _num_qudits = 2
     _num_params = 1
     _qasm_name = 'rxx'
 
+    def __init__(self, num_levels: int = 2, level_1: int = 0, level_2: int = 1, level_3: int = 0, level_4: int = 1):
+        if num_levels < 2 or not is_integer(num_levels):
+            raise ValueError(
+                'XGate num_levels must be a postive integer greater than or equal to 2.',
+            )
+        self.num_levels = num_levels
+        if level_1 > num_levels or level_2 > num_levels or level_3 > num_levels or level_4 > num_levels:
+            raise ValueError(
+                'XGate indices must be equal or less to the number of levels.',
+            )
+        self.level_1 = level_1
+        self.level_2 = level_2
+        self.level_3 = level_3
+        self.level_4 = level_4
+
     def get_unitary(self, params: RealVector = []) -> UnitaryMatrix:
         """Return the unitary for this gate, see :class:`Unitary` for more."""
         self.check_parameters(params)
 
         cos = np.cos(params[0] / 2)
         sin = -1j * np.sin(params[0] / 2)
 
-        return UnitaryMatrix(
-            [
-                [cos, 0, 0, sin],
-                [0, cos, sin, 0],
-                [0, sin, cos, 0],
-                [sin, 0, 0, cos],
-            ],
-        )
+        matrix = np.eye(self.num_levels, dtype=np.complex128)
+        matrix[
+            self.level_1 * self.num_levels + self.level_3,
+            self.level_1 * self.num_levels + self.level_3,
+        ] = cos
```

Comment:
this can be reformated from:
```
        matrix[
            self.level_1 * self.num_levels + self.level_3,
            self.level_1 * self.num_levels + self.level_3,
        ] = cos
```

to
```
        idx = self.level_1 * self.num_levels + self.level_3
        matrix[idx, idx] = cos
```